### DISCUSSION TITLE
exported nuget package with static library does not add include dirs to project

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/vcpkg#19188] [Ideas]
Hi,

I created a nuget package using: 

  `./vcpkg export --nuget libxml2:x86-windows-static boost-algorithm --output=i18nlib-deps --nuget-version=1.0.1`

And installed it in my project using:

  `Install-Package i18nlib-deps -Source "C:\vcpkg"`

Unfortunately, when I then compile my project, VC++ reports an error that the libxml2 header files that I use cannot be found.

But it works correctly when I instead build and install the nuget package using:

  `./vcpkg export --nuget libxml2 boost-algorithm --output=i18nlib-deps --nuget-version=1.0.1`

And in my project:

 `Install-Package i18nlib-deps -Source "C:\vcpkg"`

(having of course uninstalled the nuget package with the static libxml2 first.)

I had disabled vcpkg system integration prior to doing any of this.

**Comment by NN---** [comment]
You need to install both "-static" and regular package.
Then you export using regular name without "-static":

```
./vcpkg export --nuget libxml2:x86-windows boost-algorithm --output=i18nlib-deps --nuget-version=1.0.1
```

IMO vcpkg should not allow such export.

**Comment by NN---** [comment]
@eundlswe Pay attention to this:
https://blogs.msdn.microsoft.com/vcblog/2016/11/01/vcpkg-updates-static-linking-is-now-available/

You must manually change vcxproj if you want static to work correctly.

```xml
 <PropertyGroup Label="Globals">
 <!-- .... -->
 <VcpkgTriplet Condition="'$(Platform)'=='Win32'">x86-windows-static</VcpkgTriplet>
 <VcpkgTriplet Condition="'$(Platform)'=='x64'">x64-windows-static</VcpkgTriplet>
</PropertyGroup>
```

**Comment by NancyLi1013** [comment]
@eundlswe , please do as the above description that NN-- commented..
Thanks.

**Comment by NN---** [comment]
@NancyLi1013 Is there a solution for this ?
I would expect installing one, would install the other and manual vcxproj change is not a nice solution

**Comment by NancyLi1013** [comment]
@NN--- 

Sorry, there is no better solution for this. It is still a bug that needs to be fixed.
